test/controllers/beneficial.owner.gov.controller.spec.ts

```diff
@@ -460,13 +460,49 @@ describe("BENEFICIAL OWNER GOV controller", () => {
       expect(resp.text).toContain(BENEFICIAL_OWNER_GOV_PAGE_HEADING);
       expect(resp.text).not.toContain(ErrorMessages.ENTER_DATE);
       expect(resp.text).not.toContain(ErrorMessages.DAY);
-      expect(resp.text).not.toContain(ErrorMessages.MONTH);
+      expect(resp.text).toContain(ErrorMessages.MONTH);
       expect(resp.text).not.toContain(ErrorMessages.YEAR);
-      expect(resp.text).toContain(ErrorMessages.INVALID_DATE);
+      expect(resp.text).not.toContain(ErrorMessages.INVALID_DATE);
+      expect(resp.text).not.toContain(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY);
+      expect(mockSaveAndContinue).not.toHaveBeenCalled();
+    });
+
+    test(`renders the current page ${config.BENEFICIAL_OWNER_GOV_PAGE} with only missing YEAR error when year is zero`, async () => {
+      const beneficialOwnerGov = { ...REQ_BODY_BENEFICIAL_OWNER_GOV_FOR_START_DATE_VALIDATION };
+      beneficialOwnerGov["start_date-day"] = "30";
+      beneficialOwnerGov["start_date-month"] = "11";
+      beneficialOwnerGov["start_date-year"] = "00";
+      const resp = await request(app)
+        .post(config.BENEFICIAL_OWNER_GOV_URL)
+        .send(beneficialOwnerGov);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_GOV_PAGE_HEADING);
+      expect(resp.text).not.toContain(ErrorMessages.ENTER_DATE);
+      expect(resp.text).not.toContain(ErrorMessages.DAY);
+      expect(resp.text).not.toContain(ErrorMessages.MONTH);
+      expect(resp.text).toContain(ErrorMessages.YEAR);
+      expect(resp.text).not.toContain(ErrorMessages.INVALID_DATE);
       expect(resp.text).not.toContain(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY);
       expect(mockSaveAndContinue).not.toHaveBeenCalled();
     });
 
+    test(`leading zeros are stripped from start date`, async () => {
+      mockPrepareData.mockReturnValueOnce(BENEFICIAL_OWNER_GOV_OBJECT_MOCK);
+      const beneficialOwnerGov = { ...BENEFICIAL_OWNER_GOV_BODY_OBJECT_MOCK_WITH_ADDRESS };
+      beneficialOwnerGov["start_date-day"] = "0030";
```

Comment:
Do you need to add a test (for example) when leading zeroes feature in a non-valid day, month or year. E.g. on this line, what happens when `start_date-day` is "0032"?

Reply:
just tested this and eg start date of 0032 still returns 'invalid date' error even when the "0" stripping is not added. So we will still get the same error if entering 0032 whether it strips the zeroes or not. So i don't think adding tests for those scenarios will prove anything.

Reply:
We can test the year length as 0040 will fail validation with "0" trimming (length = 2 when validated) and will pass validation without "0" trimming as length will be 4 when validated.

Reply:
Sounds good to me. The only reason for adding tests in these scenarios was because the scenario wasn't tested at all and if we just assert on what it does now, we'll then pick up any potential breakages or regressions in the future (when the date logic changes again...).

Reply:
ok i can add the extra tests , won't hurt